tests/CPT_test.py

```diff
@@ -0,0 +1,332 @@
+                + eos
+            )
+
+            # determine label tokens, to calculate loss only over them when labels_loss == True
+            attention_mask = [1] * len(input_ids)
+            input_type_mask = (
+                [1] * len(input_template_tokenized_part1)
+                + [2] * len(input_tokenized)
+                + [1] * len(input_template_tokenized_part2)
+                + [0] * len(sep_tokenized)
+                + [3] * len(label_template_part1_tokenized)
+                + [4] * len(label_tokenized)
+                + [3] * len(label_template_part2_tokenized)
+                + [0] * len(eos)
+            )
+
+            assert len(input_type_mask) == len(input_ids) == len(attention_mask)
+
+            return input_ids, attention_mask, input_type_mask
+
+        def __len__(self):
+            return len(self.input_ids)
+
+        def __getitem__(self, idx):
+            return {
+                "input_ids": self.input_ids[idx],
+                "attention_mask": self.attention_mask[idx],
+                "input_type_mask": self.input_type_mask[idx],
+            }
+
+    dataset = CPTDataset(data, tokenizer, TEMPLATE)
+
+    return dataset
+
+
+def test_model_initialization_text(global_tokenizer, config_text):
+    """Test model loading and PEFT model initialization."""
+    base_model = AutoModelForCausalLM.from_pretrained(MODEL_NAME, cache_dir=".", trust_remote_code=True)
+
+    model = get_peft_model(base_model, config_text)
+    assert model is not None, "PEFT model initialization failed"
+
+
+def test_model_initialization_random(global_tokenizer, config_random):
+    """Test model loading and PEFT model initialization."""
+    base_model = AutoModelForCausalLM.from_pretrained(MODEL_NAME, cache_dir=".", trust_remote_code=True)
+
+    model = get_peft_model(base_model, config_random)
+    assert model is not None, "PEFT model initialization failed"
+
+
+def test_model_training_random(sst_data, global_tokenizer, collator, config_random):
+    """Perform a short training run to verify the model and data integration."""
+
+    base_model = AutoModelForCausalLM.from_pretrained(MODEL_NAME, cache_dir=".", trust_remote_code=True)
+    model = get_peft_model(base_model, config_random)
+    emb = model.prompt_encoder.default.embedding.weight.data.clone().detach()
+    training_args = TrainingArguments(
+        output_dir="./results",
+        per_device_train_batch_size=1,
+        num_train_epochs=2,
+        remove_unused_columns=False,
+        save_strategy="no",
+        logging_steps=1,
+    )
+
+    train_dataset = dataset(sst_data["train"], global_tokenizer)
+
+    trainer = Trainer(model=model, args=training_args, train_dataset=train_dataset, data_collator=collator)
+
+    try:
+        trainer.train()
+    except Exception as e:
+        pytest.fail(f"Training failed with error: {e}")
+
+    assert torch.all(model.prompt_encoder.default.embedding.weight.data.clone().detach().cpu() == emb.cpu())
```

Comment:
For my understanding, this is to test that these embeddings are frozen? Let's add a comment.